### PR TITLE
fix version parsing

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

resolving some compatibility issue with aplha versions

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/utilities/imports.py [hunk 1]
```diff
@@ -54,13 +54,13 @@ def _compare_version(package: str, op, version) -> bool:
     except (ModuleNotFoundError, DistributionNotFound):
         return False
     try:
-        pkg_version = LooseVersion(pkg.__version__)
+        pkg_version = Version(pkg.__version__)
```
Comment (carmocca) [inline pytorch_lightning/utilities/imports.py:57]: Can you add some tests for the version numbers that require these changes?

File: CHANGELOG.md [hunk 2]
```diff
@@ -264,6 +264,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed process rank not being available right away after `Trainer` instantiation ([#6941](https://github.com/PyTorchLightning/pytorch-lightning/pull/6941))
 
 
+- Fixed parsing some pre-release version ([#6999](https://github.com/PyTorchLightning/pytorch-lightning/pull/6999))
```
Comment (carmocca) [inline CHANGELOG.md:267]: This could use a better message

Comment (Borda) [reply CHANGELOG.md:267]: any suggestion? :]

File: pytorch_lightning/utilities/imports.py (hunk 1 above)
Comment (Borda) [reply pytorch_lightning/utilities/imports.py:57]: not really, this function is loading the version from the system so patch/mock does not work...
```python
@mock.patch("torch.__version__", return_value="1.8.0a19")
def test_compare_version(torch):
    assert _compare_version("torch", operator.ge, "1.7.1")
    assert not _compare_version("torch", operator.ge, "1.8.0")
```
cc: @awaelchli

File: CHANGELOG.md (hunk 2 above)
Comment (carmocca) [reply CHANGELOG.md:267]: Fixed parsing for pre-release package versions?

**Review by justusschock — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

**Review by awaelchli — APPROVED** [review]

**Comment by kaushikb11** [comment]
@Borda Any update on this?

File: pytorch_lightning/utilities/imports.py [hunk 3]
```diff
@@ -54,13 +54,11 @@ def _compare_version(package: str, op, version) -> bool:
     except (ModuleNotFoundError, DistributionNotFound):
         return False
     try:
-        pkg_version = LooseVersion(pkg.__version__)
-    except AttributeError:
-        return False
-    if not (hasattr(pkg_version, "vstring") and hasattr(pkg_version, "version")):
+        pkg_version = Version(pkg.__version__)
+    except TypeError:
```
Comment (ananthsub) [inline pytorch_lightning/utilities/imports.py:58]: @Borda this breaks if the pkg doesn't have `__version__` defined. the previously caught attribute error is now raised. i'm seeing this with torchtext which doesn't seem to export the version info from source: https://github.com/pytorch/text/blob/ccebe7ced8d2229b98e39546fcfbf3e233bd70d9/torchtext/__init__.py#L10-L14

cc @parmeet

Comment (Borda) [reply pytorch_lightning/utilities/imports.py:58]: Well, this seems as `torchtext` as they import own module `version` which is missing... Also their `git_version` is gone

**Pull request merged**